pkg/cluster/install.go

```diff
@@ -131,15 +130,20 @@ func (m *manager) Update(ctx context.Context) error {
 		steps.Action(m.configureIngressCertificate),
 		steps.Action(m.updateOpenShiftSecret),
 		steps.Action(m.updateAROSecret),
-		// Hive reconciliation: we mostly need it to make sure that
-		// hive has the latest credentials after rotation.
-		steps.Action(m.hiveCreateNamespace),
-		steps.Action(m.hiveEnsureResources),
-		steps.Condition(m.hiveClusterDeploymentReady, 5*time.Minute, true),
-		steps.Action(m.hiveResetCorrelationData),
 	}
 
-	return m.runSteps(ctx, steps, false)
+	if m.adoptViaHive {
```

Comment:
why are we adding `adoptViaHive` into the `az aro update` customer workflow?

Reply:
I think it's because hive also requires the cluster service principal to function properly.  So if a customer goes through the Update flow to rotate credentials and their cluster is already adopted by hive, it needs to rotate the credentials within Hive too.

This does have a side-affect of adopting the cluster to hive on an update if the cluster isn't already adopted.

Reply:
:+1: